Modules/IO/JPEG/src/itkJPEGImageIO.cxx

```diff
@@ -508,29 +507,26 @@ JPEGImageIO::WriteSlice(std::string & fileName, const void * const buffer)
   // start compression
   jpeg_start_compress(&cinfo, TRUE);
 
-  volatile const JSAMPLE * outPtr = ((const JSAMPLE *)buffer);
-
-  // write the data. in jpeg, the first row is the top row of the image
-  auto *    row_pointers = new JSAMPROW[height];
-  const int rowInc = numComp * width;
-  for (unsigned int ui = 0; ui < height; ++ui)
   {
-    row_pointers[ui] = const_cast<JSAMPROW>(outPtr);
-    outPtr = const_cast<JSAMPLE *>(outPtr) + rowInc;
+    const unsigned int rowbytes = numComp * width;
+    unsigned int       count_rows = 0;
+    while (cinfo.next_scanline < cinfo.image_height)
```

Comment:
This should avoid allocating almost an entire image worth of temporary memory, correct? Just 1 temp scanline instead of all of them?

Reply:
Updated, now there is still _while_ loop, but almost always will run in 1 iteration, very close as it was before, but safer if the lib will not be able to write all lines at once. The reason for change was that there was possible memory leak, pointer was allocated after _setjmp_ point.

> This should avoid allocating almost an entire image
Allocation is not big: image height * size of pointer

P.S.
Previous version, with 1 line per call, was also correct, all reference apps doing so, even if JPEG recommends to do less calls. But probably now is better.